ethstats/ethstats.go

```diff
@@ -185,10 +185,18 @@ func (s *Service) loop() {
 	// Subscribe to chain events to execute updates on
 	chainHeadCh := make(chan core.ChainHeadEvent, chainHeadChanSize)
 	headSub := s.backend.SubscribeChainHeadEvent(chainHeadCh)
+	if headSub == nil {
+		log.Info("Stats daemon stopped due to nil head subscription")
+		return
+	}
```

Comment:
It's easy to avoid this error subscribing earlier. If you create the subscriptions in `New`, they can never be nil.

Reply:
If they're created in `Start`, then the deferred `Unsubscribe` will become "decoupled" from the Subscribe, which is not great either...

Reply:
The Unsubscribe would need to move into Service.Stop then.